src/classes/UTIL_BatchJobService.cls

```diff
@@ -0,0 +1,220 @@
+        public Boolean isInProgress = false;
+
+        /***
+        * @description Constructor
+        * @param job AsyncApexJob record
+        */
+        public BatchJob(AsyncApexJob job) {
+            if (job == null) {
+                throw new BatchException(BATCH_JOB_REQUIRED);
+            }
+
+            batchId = job.Id;
+            status = job.Status;
+            extendedStatus = job.ExtendedStatus;
+            jobItemsProcessed = job.JobItemsProcessed;
+            totalJobItems = job.TotalJobItems;
+            numberOfErrors = job.NumberOfErrors;
+
+            createdDate = job.CreatedDate;
+            completedDate = job.CompletedDate;
+
+            timeElapsed = calcTimeElapsed(createdDate, completedDate);
+            percentComplete = calcPercentComplete(jobItemsProcessed, totalJobItems, status);
+            isInProgress = isInProgress(status);
+        }
+
+        /***
+        * @description Calculates time elapsed since the batch job has started. 
+        * 
+        * @param startDate Batch Start Date
+        * @param endDate When completed, the endDate is completion date; 
+        * otherwise, the endDate is the current datetime.
+        * @return String Time elapsed
+        */
+        @TestVisible
+        private String calcTimeElapsed(DateTime startDate, DateTime endDate) {
+            endDate = endDate == null ? Datetime.now() : endDate;
+
+            Long milliseconds = endDate.getTime() - startDate.getTime();
+            Long days = milliseconds / 1000 /60 /60 /24;
+
+            DateTime diffDate = DateTime.newInstance(milliseconds);
+
+            return (days > 0 ? String.valueOf(days) + ' ' + System.Label.BatchProgressTimeElapsedDays + ' ' : '') +
+                String.format(
+                    '{0}:{1}:{2}', 
+                    new String[] { formatTime(diffDate.hourGmt()), formatTime(diffDate.minuteGmt()), formatTime(diffDate.secondGmt()) }
+                );
+        }
+
+        /***
+        * @description Formats hours, minutes and seconds of the time
+        * @param value Time
+        * @return String Formatted time
+        */
+        private String formatTime(Long value) {
+            return String.valueOf(value).leftPad(2, '0').right(2);
+        }
+
+        /***
+        * @description Calculates the percentage of the batch job completion
+        * @param jobItemsProcessed Processed batch iterations
+        * @param totalJobItems Total batch iterations
+        * @param status Batch job status
+        * @return Long Percentage completed
+        */
+        @TestVisible
+        private Long calcPercentComplete(Integer jobItemsProcessed, Integer totalJobItems, String status) {
+            final Integer defaultPercentComplete = 5;
+
+            Long percentComplete = defaultPercentComplete;
+
+            if (ApexJobStatus.COMPLETED.name() == status) {
+                percentComplete = 100;
+
+            } else if (ApexJobStatus.QUEUED.name() == status) {
```

Comment:
Holding should be in here as well, basically the same as Queued.

Reply:
Thank you @force2b ! Changes have been applied.